libsolidity/parsing/Parser.cpp

```diff
@@ -361,12 +361,22 @@ Parser::FunctionHeaderParserResult Parser::parseFunctionHeader(bool _forceEmptyN
 		{
 			if (result.visibility != Declaration::Visibility::Default)
 			{
-				parserError(string(
-					"Visibility already specified as \"" +
-					Declaration::visibilityToString(result.visibility) +
-					"\"."
-				));
-				m_scanner->next();
+				// Is it part of function definition? Only a heuristic; will fail for more than 2 visibility modifiers
+				if (
+					m_scanner->peekNextToken() == Token::LBrace ||
+					Token::isStateMutabilitySpecifier(m_scanner->peekNextToken())
+				)
+				{
+					parserError(string(
+						"Visibility already specified as \"" +
+						Declaration::visibilityToString(result.visibility) +
+						"\"."
+					));
+					m_scanner->next();
+				}
+				else
+					// Part of function type, so make this token a part of variable declaration.
+					break;
```

Comment:
I think it would be more readable if the condition is inverted and `break` is next to an if statement.

Reply:
Never mind, the same pattern is used in this function already.